x/wasm/migrations/v3/store_test.go

```diff
@@ -0,0 +1,135 @@
+
+	// contract with any addresses permission
+	anyAddrPermission := v3.AccessConfig{
+		Permission: v3.AccessTypeAnyOfAddresses,
+		Addresses:  []string{creator.String()},
+	}
+	contract2 := keeper.StoreRandomContract(t, ctx, keepers, &mock)
+	newInfo2 := wasmKeeper.GetCodeInfo(ctx, contract2.CodeID)
+	legacyInfo2 := v3.CodeInfo{
+		CodeHash:          newInfo2.CodeHash,
+		Creator:           newInfo2.Creator,
+		InstantiateConfig: anyAddrPermission,
+	}
+
+	store.Set(types.GetCodeKey(contract2.CodeID), cdc.MustMarshal(&legacyInfo2))
+
+	// contract with everybody permission
+	allowEverybodyPermission := v3.AccessConfig{
+		Permission: v3.AccessTypeEverybody,
+	}
+	contract3 := keeper.StoreRandomContract(t, ctx, keepers, &mock)
+	newInfo3 := wasmKeeper.GetCodeInfo(ctx, contract3.CodeID)
+	legacyInfo3 := v3.CodeInfo{
+		CodeHash:          newInfo3.CodeHash,
+		Creator:           newInfo3.Creator,
+		InstantiateConfig: allowEverybodyPermission,
+	}
+
+	store.Set(types.GetCodeKey(contract3.CodeID), cdc.MustMarshal(&legacyInfo3))
+
+	// contract with nobody permission
+	allowNobodyPermission := v3.AccessConfig{
+		Permission: v3.AccessTypeNobody,
+	}
+	contract4 := keeper.StoreRandomContract(t, ctx, keepers, &mock)
+	newInfo4 := wasmKeeper.GetCodeInfo(ctx, contract4.CodeID)
+	legacyInfo4 := v3.CodeInfo{
+		CodeHash:          newInfo4.CodeHash,
+		Creator:           newInfo4.Creator,
+		InstantiateConfig: allowNobodyPermission,
+	}
+
+	store.Set(types.GetCodeKey(contract4.CodeID), cdc.MustMarshal(&legacyInfo4))
+
+	// set only address permission params
+	ps := v3.Params{
+		CodeUploadAccess: v3.AccessConfig{
+			Permission: v3.AccessTypeOnlyAddress,
+			Address:    creator.String(),
+		},
+		InstantiateDefaultPermission: v3.AccessTypeOnlyAddress,
+	}
+
+	bz, err := cdc.Marshal(&ps)
+	require.NoError(t, err)
+	store.Set(types.ParamsKey, bz)
+
+	// when
+	err = keeper.NewMigrator(*wasmKeeper, nil).Migrate3to4(ctx)
+
+	// then
+	require.NoError(t, err)
+
+	expParams := types.Params{
+		CodeUploadAccess:             types.AccessTypeAnyOfAddresses.With(creator),
+		InstantiateDefaultPermission: types.AccessTypeAnyOfAddresses,
+	}
+
+	// params are migrated
+	assert.Equal(t, expParams, wasmKeeper.GetParams(ctx))
+
+	// access config for only address is migrated
+	info1 := wasmKeeper.GetCodeInfo(ctx, contract1.CodeID)
+	assert.Equal(t, types.AccessTypeAnyOfAddresses.With(creator), info1.InstantiateConfig)
+
+	// access config for any addresses is not migrated
+	info2 := wasmKeeper.GetCodeInfo(ctx, contract2.CodeID)
+	assert.Equal(t, types.AccessTypeAnyOfAddresses.With(creator), info2.InstantiateConfig)
+
+	// access config for allow everybody is not migrated
+	info3 := wasmKeeper.GetCodeInfo(ctx, contract3.CodeID)
+	assert.Equal(t, types.AllowEverybody, info3.InstantiateConfig)
+
+	// access config for allow nobody is not migrated
+	info4 := wasmKeeper.GetCodeInfo(ctx, contract4.CodeID)
+	assert.Equal(t, types.AllowNobody, info4.InstantiateConfig)
+}
```

Comment:
Good test cases! 
Please consider table test in the future to make it more readable but no need to change this now.